csrc/python_frontend/fusion_definition.cpp

```diff
@@ -231,6 +243,14 @@ void FusionDefinition::finalizeSchedule(
   user_sched_->runtime_info.reset();
   prev_fusion_ = nullptr;
   user_sched_ = nullptr;
+
+  // Scheduling the fusion can add states to recording_state.
+  // Remove the schedule-only states.
+  size_t num_states_to_remove =
+      recording_state_.size() - num_recording_states_presched_;
+  for (size_t rnd = 0; rnd < num_states_to_remove; ++rnd) {
+    recording_state_.pop_back();
+  }
```

Comment:
another naive question: should this be done in `setupSchedule` instead?

My understanding is that, if I'm playing with this without using a context manager for scheduling. I could potentially create new tensor with things like

`t = fd.sched.rfactor(...)`

If I manually call `fd._finalize_schedule()`, the `t` would vanish afterwards, even though I still have a handle in python.

Reply:
Ah. Python where you can access anything, anywhere, anytime.

Reply:
Okay. Moved that to `setupSchedule`. I also removed resetting `user_sched_` to `nullptr` in `finalizeSchedule`, so users can print user schedule afterwards.